src/LaunchDarkly.EventSource/IEventSource.cs

```diff
@@ -54,25 +86,90 @@ public interface IEventSource
         Task StartAsync();
 
         /// <summary>
-        /// Triggers the same "close and retry" behavior as if an error had been encountered on the stream.
+        /// Attempts to receive a message from the stream.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// If the stream is not already active, this calls <see cref="StartAsync"/> to
+        /// establish a connection.
+        /// </para>
+        /// <para>
+        /// As long as the stream is active, the method waits until a message is available.
+        /// If the stream fails, the default behavior is to throw a <see cref="StreamException"/>,
+        /// but you can configure an <see cref="ErrorStrategy"/> to allow the client to retry
+        /// transparently instead. However, the client will never retry if you called
+        /// <see cref="Close"/> or <see cref="IDisposable.Dispose"/>; in that case, trying to
+        /// read will always throw a <see cref="StreamClosedByCallerException"/>.
+        /// </para>
+        /// <para>
+        /// The returned <see cref="MessageEvent"/> may contain references to temporary internal
+        /// data that will be overwritten if you read another event. If you are going to store
+        /// the event somewhere past the point where you read another event, you should call
+        /// <see cref="MessageEvent.ReadFully"/> to obtain a long-lived copy.
+        /// </para>
+        /// <para>
+        /// This method must be called from the same thread that first started using the
+        /// stream (that is, the thread that called <see cref="StartAsync"/> or read the
+        /// first event).
+        /// </para>
+        /// </remarks>
+        /// <returns>an SSE message</returns>
+        /// <seealso cref="ReadAnyEventAsync"/>
+        Task<MessageEvent> ReadMessageAsync();
```

Comment:
One notable API difference between this and the Java `okhttp-eventsource` is that the latter also has methods that return messages/events as an `Iterable` (Java equivalent of an `IEnumerable`), so you can just treat them like any collection instead of explicitly asking for each event. That's convenient, but the problem is that _async_ generators are a C# 8.0 feature, and we're still constrained to use C# 7.3 in order to maintain .NET Standard 2.0 compatibility.